### DISCUSSION TITLE
Invalidating multiple resource identifiers from same live list queries

### DISCUSSION THREAD

**Discussion by strblr** [n1ru4l/graphql-live-query#914] [Q&A] [answered]
Hello,

When invalidating multiple resource identifiers that are all included in the response payload of the same (live) list query, for example:

```ts
invalidate(["Todo:1", "Todo:2", "Todo:3", "Todo:4"])
```

Does it trigger 4 re-executions of (let's say) `Query.todos` if the previous data for that query included the 4 resources?
If not, what mechanism guarantees that it doesn't?

Thanks!

**Comment by n1ru4l** [comment, marked as answer]
The [`ResourceTracker`](https://github.com/n1ru4l/graphql-live-query/blob/17f57100f583d170c51e71c012cd83b20b0d4023/packages/in-memory-live-query-store/src/ResourceTracker.ts#L10) returns a [set of identifiers that only includes each record once](https://github.com/n1ru4l/graphql-live-query/blob/17f57100f583d170c51e71c012cd83b20b0d4023/packages/in-memory-live-query-store/src/ResourceTracker.ts#L72-L84).

Thus, when invoking `liveQueryStore.invalidate(["Todo:1", "Todo:2", "Todo:3", "Todo:4"])` a single record is only scheduled for re-executed once.

**Reply by strblr** [reply]
Makes sense, thanks for the quick response!
